cli/azd/pkg/apphost/generate.go

```diff
@@ -1025,16 +1292,42 @@ func (b *infraGenerator) buildEnvBlock(env map[string]string, manifestCtx *genCo
 		// connectionString detection:
 		//  - If the env-key contains "ConnectionStrings__" or the value contains "{{ connectionString" then it is considered
 		//    as a secret and added to the secrets map.
-		if strings.Contains(k, "ConnectionStrings__") || strings.Contains(resolvedValue, "{{ connectionString") {
+		if strings.Contains(k, "ConnectionStrings__") ||
+			strings.Contains(resolvedValue, "{{ connectionString") ||
+			strings.Contains(resolvedValue, "{{ secured-parameter ") {
+			if strings.Contains(resolvedValue, "{{ secretOutput ") {
+				if isComplexExp, _ := isComplexExpression(resolvedValue); !isComplexExp {
```

Comment:
I don't know if this needs to be robust, but I can see some possible conditions that may not be handled explicitly:

- `isComplexExpression` could return true if the value is neither complex nor simple (the parse fails), so that case is not handled separately
- Not clear from the code, but we remove `{{ secretOutput ` or `{{ secured-parameter ` in some cases -- what happens if it's not removed?

I don't know if we need to clean anything up for this PR, but if so, maybe just leave comments on what expected inputs are like.

Reply:
> isComplexExpression could return true if the value is neither complex nor simple (the parse fails), so that case is not handled separately

What do you mean with `the parse fails` ?